web/src/components/pages/dashboard/challenge/weekly-challenge.tsx

```diff
@@ -13,132 +18,195 @@ const WEEKS: Array<string> = [
+                    title={WEEKS[this.currentWeek + 1]}
+                    week={this.currentWeek + 1}
+                    isNarrow={isNarrow}
+                    avatarUrl={user.account.avatar_url}
+                    logoUrl={challengeLogoUrls[user.account.enrollment.team]}
+                    weekly={weekly}
+                  />
+                ) : (
+                  this.pastWeek.map(
+                    (value, index) =>
+                      weekly && (
+                        <WeeklyChallengeBoard
+                          isDisabled
+                          title={WEEKS[value]}
+                          week={value}
+                          key={index}
+                          isNarrow={isNarrow}
+                          avatarUrl={user.account.avatar_url}
+                          logoUrl={
+                            challengeLogoUrls[user.account.enrollment.team]
+                          }
+                          weekly={weekly}
+                        />
+                      )
+                  )
+                )}
+              </div>
+            )}
+            <div>
+              <p className="weekly-title">{this.label} challenge</p>
+              {this.label === 'Future'
+                ? weekly && (
                     <WeeklyChallengeBoard
                       isDisabled
-                      title={WEEKS[value]}
-                      week={value}
-                      key={index}
+                      title={WEEKS[this.currentWeek + 2]}
+                      week={this.currentWeek + 2}
                       isNarrow={isNarrow}
-                      avatarUrl={account.avatar_url}
-                      logoUrl={challengeLogoUrls[account.enrollment.team]}
+                      avatarUrl={user.account.avatar_url}
+                      logoUrl={challengeLogoUrls[user.account.enrollment.team]}
                       weekly={weekly}
                     />
                   )
-              )}
+                : this.pastWeek.map(
+                    (value, index) =>
+                      weekly && (
+                        <WeeklyChallengeBoard
+                          isDisabled
+                          title={WEEKS[value]}
+                          week={value}
+                          key={index}
+                          isNarrow={isNarrow}
+                          avatarUrl={user.account.avatar_url}
+                          logoUrl={
+                            challengeLogoUrls[user.account.enrollment.team]
+                          }
+                          weekly={weekly}
+                        />
+                      )
+                  )}
+            </div>
+          </div>
         </div>
       </div>
-    </div>
-  );
+    );
+  }
 }
+
+export default connect<PropsFromState>(
+  ({ api, user, notifications }: StateTree) => ({
+    api,
+    user,
+    notifications,
+  }),
+  null,
+  null,
+  { forwardRef: true }
+)(WeeklyChallengeCard);
```

Comment:
Note: I didn't look _too_ closely at this file during review, since many of the changes seem to be formatting / not directly related to toast notifications.

Reply:
Yep, correct. These parts are many formatting changes.